_studio/mfx_lib/encode_hw/hevc/linux/g9/hevcehw_g9_dirty_rect_lin.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2020 Intel Corporation
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#pragma once
+
+#include "mfx_common.h"
+#if defined(MFX_ENABLE_H265_VIDEO_ENCODE) && defined(MFX_VA_LINUX)
+
+#include "hevcehw_g9_dirty_rect.h"
+#include "va/va.h"
+
+namespace HEVCEHW
+{
+namespace Linux
+{
+namespace Gen9
+{
+class DirtyRect
+    : public HEVCEHW::Gen9::DirtyRect
+{
+public:
+
+    DirtyRect(mfxU32 FeatureId)
+        : HEVCEHW::Gen9::DirtyRect(FeatureId)
+    {}
+
+protected:
+    virtual void InitAlloc(const FeatureBlocks& /*blocks*/, TPushIA Push) override;
+
+    std::vector<VARectangle> m_vaDirtyRects = std::vector<VARectangle>(MAX_NUM_DIRTY_RECT);;
```

Comment:
Correct way would be ```std::vector<VARectangle> m_vaDirtyRects(MAX_NUM_DIRTY_RECT);``` 

But do we really need such preallocation?

Reply:
std::vector<VARectangle> m_vaDirtyRects(MAX_NUM_DIRTY_RECT);
It doesn't work.